.github/actions/setup-arc-e2e/action.yaml

```diff
@@ -0,0 +1,55 @@
+name: 'Setup ARC E2E Test Action'
+description: 'Build controller image, create kind cluster, load the image, and exchange ARC configure token.'
+
+inputs:
+  github-app-id:
+    description: 'GitHub App Id for exchange access token'
+    required: true
+  github-app-pk:
+    description: "GitHub App private key for exchange access token"
+    required: true
+  github-app-org:
+    description: 'The organization the GitHub App has installed on'
+    required: true
+  docker-image-name:
+    description: "Local docker image name for building"
+    required: true
+  docker-image-tag:
+    description: "Tag of ARC Docker image for building"
+    required: true
+
+outputs:
+  token:
+    description: 'Token to use for configure ARC'
+    value: ${{steps.config-token.outputs.token}}
+
+runs:
+  using: "composite"
+  steps:
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v2
+      with:
+        version: latest
+
+    - name: Docker build test image
+      shell: bash
+      run: |
+          export DOCKER_IMAGE_NAME=${{inputs.docker-image-name}}
+          export VERSION=${{inputs.docker-image-tag}}
+          export IMG_RESULT=load
+          make docker-buildx
```

Comment:
While I think it's important for us to stay consistent, I really think we should optimise this step & cache some of the image layers. This test will run frequently and this will be the biggest bottleneck in the flow

If you use something like this: https://github.com/actions-runner-controller/releases/blob/222166eca744b5a519d65428cef8f7eb471fb3e6/.github/workflows/publish-canary.yaml#L63

We'll be able to speed up our runtime substantially

Reply:
I don't want to give these test workflows any `write` permission anything to the registry.

Reply:
@TingluoHuang you don't have to. The action will use our cache service https://github.com/moby/buildkit#github-actions-cache-experimental not packages.

Reply:
Updating...